src/ros_can/test/test_can.cpp

```diff
@@ -10,11 +10,77 @@
 
 /**
  * @test This test case checks if the control callback function correctly writes the steering and
- * throttle commands to the CAN bus.
+ * throttle commands to the CAN bus. 
+ * 
+ * Scenario: Steering and Throttle between the limits
  */
 TEST_F(RosCanTest, ControlCallback) {
-  prepare_out_of_range_values(0.3, 0.3, 1);
-    ros_can_->control_callback(control_command_);
+  prepare_control_publish_values(0.3, 0.3, 1);
+  ros_can_->control_callback(control_command_);
+}
+
+/**
+ * @test Test Scenario with throttle below the limits.
+ */
+TEST_F(RosCanTest, ControlCallbackLowerThrottle) {
+  prepare_control_publish_values(-1.1, 0.3, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+/**
+ * @test Test Scenario with throttle above the limits.
+ */
+TEST_F(RosCanTest, ControlCallbackHigherThrottle) {
+  prepare_control_publish_values(1.1, 0.3, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+/**
+ * @test Test Scenario with steering below the limits.
+ */
+TEST_F(RosCanTest, ControlCallbackLowerSteering) {
+  prepare_control_publish_values(0.3, -0.5, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+/**
+ * @test Test Scenario with steering above the limits.
+ */
+TEST_F(RosCanTest, ControlCallbackHigherSteering) {
+  prepare_control_publish_values(0.3, 0.5, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+
+/**
+ * @test This test represesnts the case where the steering ot the throttle violates the expected limits (Higher Value). 
+ * Expected message to not be written to CAN
+ * 
+ */
+TEST_F(RosCanTest, TestOutOfRangeUpperSteeringThrottle) {
+  prepare_control_publish_values(STEERING_UPPER_LIMIT + 1, STEERING_UPPER_LIMIT + 1, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+/**
+ * @test This test represesnts the case where the steering ot the throttle violates the expected limits (Lower Value). 
+ * Expected message to not be written to CAN
+ * 
+ */
+TEST_F(RosCanTest, TestOutOfRangeLowerSteeringThrottle) {
+  prepare_control_publish_values(STEERING_LOWER_LIMIT - 1, STEERING_LOWER_LIMIT - 1, 0);
+  ros_can_->control_callback(control_command_);
+}
+
+
+/**
+ * @test This test represesnts the case where the steering ot the throttle violates the expected limits (Higher and Lower Value). 
+ * Expected message to not be written to CAN
+ * 
+ */
+TEST_F(RosCanTest, TestOutOfRangeSingleSteeringThrottle) {
+  prepare_control_publish_values(STEERING_UPPER_LIMIT - 1, STEERING_UPPER_LIMIT + 1, 0);
```

Comment:
Why are we using STEERING_UPPER_LIMIT in the throttle part here?

Reply:
Just reusing the macro, just changed it!